extensions/vscode/scripts/build.mjs

```diff
@@ -0,0 +1,117 @@
+import { watch } from 'chokidar';
+import * as esbuild from 'esbuild';
+import { execa } from 'execa';
+import fs from 'node:fs';
+import { createRequire } from 'node:module';
+import { join, dirname } from 'path';
+import { Worker } from 'node:worker_threads';
+import { fileURLToPath } from 'node:url';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const require = createRequire(import.meta.url);
+const production = process.argv.includes('--production');
+const isWatch = process.argv.includes('--watch');
+
+async function main() {
+  const ctx = await esbuild.context({
+    entryPoints: {
+      extension: 'src/extension.ts',
+      server: './src/language-server/index.ts',
+    },
+    bundle: true,
+    format: 'cjs',
+    minify: production,
+    sourcemap: !production,
+    sourcesContent: false,
+    tsconfig: './tsconfig.json',
+    platform: 'node',
+    outdir: 'dist',
+    define: { 'process.env.NODE_ENV': production ? '"production"' : '"development"' },
+    external: ['vscode'],
+    plugins: [esbuildUMD2ESMPlugin],
+  });
+
+  if (isWatch) {
+    const buildMetadataSchemaDebounced = debounce(buildMetadataSchema, 100);
+    const dependencyPath = dirname(require.resolve('@tutorialkit/types'));
+
+    watch(dependencyPath).on('all', (eventName, path) => {
+      if (eventName !== 'change' && eventName !== 'add' && eventName !== 'unlink') {
+        return;
+      }
+
+      buildMetadataSchemaDebounced();
+    });
+
+    await Promise.all([
+      ctx.watch(),
+      execa('tsc', ['--noEmit', '--watch', '--preserveWatchOutput', '--project', 'tsconfig.json'], {
+        stdio: 'inherit',
+        preferLocal: true,
+      }),
+    ]);
+  } else {
+    await ctx.rebuild();
+    await ctx.dispose();
+
+    await buildMetadataSchema();
+
+    if (production) {
+      // rename name in package json to match extension name on store:
+      const pkgJSON = JSON.parse(fs.readFileSync('./package.json', { encoding: 'utf8' }));
+
+      pkgJSON.name = 'tutorialkit';
+
+      fs.writeFileSync('./package.json', JSON.stringify(pkgJSON, undefined, 2), 'utf8');
+    }
+  }
+}
+
+async function buildMetadataSchema() {
+  const schema = await new Promise((resolve) => {
+    const worker = new Worker(join(__dirname, './load-schema-worker.mjs'));
+    worker.on('message', (value) => resolve(value));
+  });
+
+  fs.mkdirSync('./dist', { recursive: true });
+  fs.writeFileSync('./dist/schema.json', JSON.stringify(schema, undefined, 2), 'utf-8');
+
+  console.log('Updated schema.json');
```

Comment:
Maybe use our logger instead that prints a success badge as well?

Reply:
The logger is located in `packages/cli/scripts`, and when adding `import * as logger from '../../../packages/cli/scripts/logger.js';` we'll get bunch of TS errors even though this is plain JS file.

Reply:
Yea ok it's fine if we just use `console.log` here 👍